### PR TITLE
Add pm2 logrotate to AMI

### PR DESCRIPTION
**Pull request by jessemyers**
Without this, pm2 logs don't rotate and eventually exhaust disk space.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,7 +20,7 @@ Builds Ubuntu AMIs using Ansible.
  4. Choose a source AMI (e.g. from [https://cloud-images.ubuntu.com/locator/ec2](), searching
     for "trusty us-east-1 hvm:ebs-ssd"):
 
-        export AWS_SOURCE_AMI=ami-2b594f41
+        export AWS_SOURCE_AMI=ami-50759d3d
```
Comment (jessemyers) [inline README.md]: There's a new base AMI available from May.

File: ansible/roles/baseline/tasks/main.yml [hunk 2]
```diff
@@ -17,6 +17,7 @@
     - iotop
     - jq=1.4-2.1~ubuntu14.04.1
     - libffi-dev
+    - libssl-dev
```
Comment (jessemyers) [inline ansible/roles/baseline/tasks/main.yml]: I was running into problems with Python installation; the root problem was actually in the bootstrapping for Ansible. This line doesn't really change anything.

File: ansible/roles/codedeploy/tasks/main.yml [hunk 3]
```diff
@@ -40,7 +40,9 @@
     dest: /tmp/codedeploy-agent.deb
 
 - name: install the codedeploy-agent
-  apt: deb=/tmp/codedeploy-agent.deb
+  apt:
```
Comment (jessemyers) [inline ansible/roles/codedeploy/tasks/main.yml]: I ran into a problem with the latest Ansible at this point; I tried to debug but ended up just skipping the latest ansible release.

File: ansible/roles/nodejs/tasks/main.yml [hunk 4]
```diff
@@ -34,3 +34,9 @@
 
 - name: make sure pm2 starts on boot
   command: pm2 startup ubuntu
+
+- name: make sure pm2 can rotate logs
```
Comment (jessemyers) [inline ansible/roles/nodejs/tasks/main.yml]: This is the actual change.

File: aws/ebs.json [hunk 5]
```diff
@@ -90,6 +90,7 @@
       "type": "ansible-local",
       "playbook_file": "ansible/packer.yml",
       "extra_arguments": [
+        "-vvv",
```
Comment (jessemyers) [inline aws/ebs.json]: This was to help debug ansible/apt bugs.

File: scripts/bootstrap.sh [hunk 6]
```diff
@@ -12,7 +12,7 @@ sudo apt-get -qy update
 sudo apt-get -qy update
 
 echo "Installing python..."
-sudo apt-get install -y python python-dev python-setuptools
+sudo apt-get install -y python python-dev python-setuptools libffi-dev libssl-dev
```
Comment (jessemyers) [inline scripts/bootstrap.sh]: We need these now to install Python stuff.

File: scripts/bootstrap.sh [hunk 7]
```diff
@@ -22,7 +22,7 @@ sudo pip install -U setuptools
 sudo pip install -U pip
 
 echo "Installing Ansible..."
-sudo pip install ansible
+sudo pip install 'ansible<2.1.0.0'
```
Comment (jessemyers) [inline scripts/bootstrap.sh]: The latest Ansible runs into the bug above.

**Comment by adamhadani** [comment]
LGTM

**Comment by srt32** [comment]
Any reason not to merge this change as is?

**Comment by jessemyers** [comment]
I thought it was. This has already been applied to our hosts.

**Pull request closed without merging**